core/cat/factory/custom_auth_handler.py

```diff
@@ -93,37 +93,53 @@ async def authorize_user_from_jwt(
         return None
 
     async def authorize_user_from_key(
-        self,
-        user_id: str,
-        api_key: str,
-        auth_resource: AuthResource,
-        auth_permission: AuthPermission,
+            self,
+            user_id: str,
+            api_key: str,
+            auth_resource: AuthResource,
+            auth_permission: AuthPermission,
     ) -> AuthUserInfo | None:
-        http_api_key = get_env("CCAT_API_KEY")
-        ws_api_key = get_env("CCAT_API_KEY_WS")
+        http_key = get_env("CCAT_API_KEY")
+        ws_key = get_env("CCAT_API_KEY_WS")
 
-        # TODOAUTH: should we consider the user_id or just give
-        #    admin permissions to all users with the right api keys?
-
-        # chatting over websocket
-        if auth_resource == AuthResource.CONVERSATION and api_key == ws_api_key:
+        if not http_key and not ws_key:
             return AuthUserInfo(
                 id=user_id,
                 name=user_id,
                 permissions=get_base_permissions()
```

Comment:
If no keys are set, I guess we can give full permissions to anybody?
In this way if there are no keys set, you cannot use most endpoints (I need to check)

Reply:
Yep, agree with that, otherwise we could differentiate by requested resource:

```python

if not http_key and not ws_key:
            return AuthUserInfo(
                id=user_id,
                name=user_id,
                permissions=get_base_permissions() if auth_resource == AuthResource.CONVERSATION else get_full_permissions()

```